testsuite/MDAnalysisTests/analysis/test_hydrogenbonds_analysis.py

```diff
@@ -139,7 +139,56 @@ def test_guess_donors(self, h):
         ref_donors = "(resname TIP3 and name OH2)"
         donors = h.guess_donors(selection='all', max_charge=-0.5)
         assert donors == ref_donors
+class TestHydrogenBondAnalysisTIP3P_GuessHyrdogens_NoTopology(object):
+    """
+    Guess the hydrogen atoms involved in hydrogen bonds using the mass and
+    partial charge of the atoms.
+    """
+
+    @staticmethod
+    @pytest.fixture(scope='class')
+    def universe():
+        return MDAnalysis.Universe(waterPSF, waterDCD)
+
+    kwargs = {
+        'donors_sel': None,
+        'hydrogens_sel': None,
+        'acceptors_sel': None,
+        'd_h_cutoff': 1.2,
+        'd_a_cutoff': 3.0,
+        'd_h_a_angle_cutoff': 120.0
+    }
+
+    @pytest.fixture(scope='class')
+    def h(self, universe):
+        h = HydrogenBondAnalysis(universe, **self.kwargs)
+        return h
+
+    def test_guess_hydrogens(self, h):
+
+        ref_hydrogens = "(resname TIP3 and name H1) or (resname TIP3 and name H2)"
+        hydrogens = h.guess_hydrogens(selection='all')
+        assert hydrogens == ref_hydrogens
+
+    def test_guess_hydrogens_min_mass(self, h):
```

Comment:
Considering these three tests are very similar in construction, would using pytest.mark.parametrize be worth it here?

Reply:
They are testing different parameters, not different parameters' values. How would you suggest to use `pytest.mark.parametrize` in this case?

Reply:
So I might be misunderstanding what you are trying to do, so ignore me if that's the case, but I think something like this should work:

```python
pytest.mark.parametrize('min_mass, max_mass, min_charge', [
        (1.05, 1.1, 0.3),
        (0.9, 0.95, 0.3),
        (0.9, 1.1, 1.0)
    ])
def test_guess_hydrogens_criteria_noreturn(self, h, min_mass, max_mass, min_charge):
    hydrogens = h.guess_hydrogens(selection='all', min_mass=min_mass, max_mass=max_mass, min_charge=min_charge)
    assert hydrogens == ""
```

So this actually brings another point, shouldn't the code raise a `ValueError` if min_mass > max_mass is provided?

Reply:
I see. I didn't like this too much because it's not immediately clear which parameter  is changing with respect to the default, but on a second thought less repetitive code is better therefore I adopted this stile.